src/Surfnet/ServiceProviderDashboard/Application/Command/Entity/DeletePublishedProductionEntityCommand.php

```diff
@@ -30,12 +30,21 @@ class DeletePublishedProductionEntityCommand implements Command
      */
     private $manageId;
 
+    /**
+     * @var string
+     * @Assert\NotBlank
+     * @Assert\Uuid
+     */
+    private $protocol;
```

Comment:
Given the way Manage works, this was the most pragmatic solution. I do feel your feedback is valid. It however will be impossible to figure out the protocol with only the manageId, as the different entity types have different API endpoints.

Reply:
Then I would still opt to handle this in the infrastructure layer because there it should be and prevents additional logic throughout all layers . You could use the repository to fetch the entity and then delete it based on the instance type. That way only some additional logic will be in the infrastructure layer and won't bubble up trough all the layers.

Reply:
We discussed this IRL. And we are on the same page. For now we're leaving this as is. Added an action to this technical debt chore: https://www.pivotaltracker.com/story/show/167312313 to address this issue at a later stage.

Reply:
:+1: